sway/input/seatop_down.c

```diff
@@ -99,30 +178,37 @@ static const struct sway_seatop_impl seatop_impl = {
 	.pointer_axis = handle_pointer_axis,
 	.tablet_tool_tip = handle_tablet_tool_tip,
 	.tablet_tool_motion = handle_tablet_tool_motion,
+	.touch_motion = handle_touch_motion,
+	.touch_up = handle_touch_up,
+	.touch_down = handle_touch_down,
 	.unref = handle_unref,
 	.end = handle_end,
 	.allow_set_cursor = true,
 };
 
 void seatop_begin_down(struct sway_seat *seat, struct sway_container *con,
-		uint32_t time_msec, double sx, double sy) {
-	seatop_begin_down_on_surface(seat, con->view->surface, time_msec, sx, sy);
+		double sx, double sy) {
+	seatop_begin_down_on_surface(seat, con->view->surface, sx, sy);
 	struct seatop_down_event *e = seat->seatop_data;
 	e->con = con;
 
 	container_raise_floating(con);
 	transaction_commit_dirty();
 }
 
+void seatop_begin_touch_down(struct sway_seat *seat, struct sway_container *con,
+		struct wlr_event_touch_down *event, double sx, double sy) {
+	seatop_begin_down(seat, con, sx, sy);
```

Comment:
Here, should I try to use seatop_begin_down or the new seatop_begin_down_on_surface ?

Reply:
Okay I used this seatop_begin_down_on_surface